src/hasher512.cpp

```diff
@@ -160,9 +160,9 @@ namespace libtorrent
 #endif
 		}
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), data.size());
+		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), std::uint32_t(data.size()));
```

Comment:
@ajax16384 as far as I can tell, this change doesn't look right. the ``#ifdef`` implies that this is in fact a call into libcrypto, which appears to be taking a ``long int``. It looks like it's the call below that end up calling the internal sha512 function, which you point out takes a ``uint32_t``.

However, I think a better solution would be to make the internal SHA512 signature match the one from libcrytpo. But in any case, I don't see how *this* call is affected.